src/store/creator/types.ts

```diff
@@ -16,7 +16,13 @@ export type CreatorItem = {
 export type CreatorState = {
     cursor: number;
     error: SerializedError | null;
+    selection: SelectionItem | null;
     message: string;
     staged: CreatorItem | null;
     status: CreatorStatus;
 };
+
+export type SelectionItem = {
```

Comment:
This seems very similar to CreatorItem. Are we mixing concerns by converting the selection into an annotation shape before it's actually created as one?